catalog/core/catalog-core-api-impl/src/test/groovy/ddf/catalog/util/impl/ResultIterableSpec.groovy

```diff
@@ -0,0 +1,372 @@
+
+        then:
+        hasNext.is true
+    }
+
+    @Unroll
+    def '''test that #nextCalls results are returned and catalog is queried #expectedQueries times
+           when next() is called #nextCalls times, page size is #pageSize and catalog returns
+           #resultSize results at a time'''(int nextCalls, int pageSize, int resultSize, int expectedQueries) {
+        setup:
+        expectedQueries * catalogFramework.query(_ as QueryRequest) >> {
+            QueryRequest queryRequest -> buildQueryResponse(queryRequest, resultSize, 5)
+        }
+
+        Query queryMock = createQueryMock(1, pageSize)
+        QueryRequest queryRequestMock = createQueryRequestMock(queryMock)
+
+        def resultIterator = new ResultIterable(catalogFramework, queryRequestMock).iterator()
+        def results = []
+
+        when:
+        1.upto(nextCalls, { results << resultIterator.next() })
+
+        then:
+        1.upto(nextCalls, { results.get(it - 1).distanceInMeters == (Double) it })
+
+        where:
+        nextCalls | pageSize | resultSize | expectedQueries
+        1         | 1        | 1          | 1
+        2         | 1        | 2          | 1
+        3         | 1        | 1          | 3
+        2         | 2        | 1          | 2
+        2         | 0        | 1          | 2
+    }
+
+    def "next() returns the proper results when start index is not 1"() {
+        setup:
+        int startIndex = 10
+        1 * catalogFramework.query(_ as QueryRequest) >> {
+            QueryRequest queryRequest -> buildQueryResponse(queryRequest, 1, 1)
+        }
+
+        Query queryMock = createQueryMock(startIndex, 1)
+        QueryRequest queryRequestMock = createQueryRequestMock(queryMock)
+
+        def resultIterator = new ResultIterable(catalogFramework.&query, queryRequestMock).iterator()
+
+        when:
+        def result = resultIterator.next()
+
+        then:
+        result.distanceInMeters == (Double) startIndex
+    }
+
+    def "next() properly pages when default page size is used"() {
+        setup:
+        def resultSize = 20
+
+        2 * catalogFramework.query(_ as QueryRequest) >> {
+            QueryRequest queryRequest -> buildQueryResponse(queryRequest, resultSize, resultSize + 5)
+        }
+
+        Query queryMock = createQueryMock(1, 0)
+        QueryRequest queryRequestMock = createQueryRequestMock(queryMock)
+
+        def resultIterator = new ResultIterable(catalogFramework, queryRequestMock).iterator()
+        def results = []
+
+        when:
+        1.upto(resultSize + 1, { results << resultIterator.next() })
+
+        then:
+        1.upto(resultSize + 1, { results.get(it - 1).distanceInMeters == (Double) it })
+    }
+
+    def "next() when number of results from catalog varies"() {
+        setup:
+        def totalResults = 6
+        3 * catalogFramework.query(_ as QueryRequest) >> {
+            QueryRequest queryRequest -> buildQueryResponse(queryRequest, 2, totalResults)
+        } >> {
```

Comment:
This seems really quite strange... but i don't think we really have our formatter for groovy enforced (or even setup) so It probably doesn't matter for now